scripts/buildorder.py

```diff
@@ -262,104 +278,133 @@ def read_packages_from_directories(directories, fast_build_mode, full_buildmode)
+            # Checking package dependencies
+            for dep in deps:
+                if dep not in pkgs_sort:
+                    # Saving the requested dependency to determine whether the package
+                    # is in a circular dependency. If a package has a circular dependency,
+                    # the requested dependency that causes the cycle will be ignored.
+                    if build_mode and isinstance(pkgs_map[dep], TermuxSubPackage):
+                        dep = pkgs_map[dep].parent.name
+                    requireds[pkg.name] = dep
+                    required = requireds[pkg.name]
+                    while not without_cyclic_dependencies and required in requireds.keys():
+                        # Checking for cyclic dependencies of a package.
+                        required = requireds[required]
+                        if required == pkg.name:
+                            break
+                    else:
+                        break
+            else:
+                yield pkg
+                pkgs_sort.append(pkg.name)
+                pkgs_sort += subpkgs
+                if pkg.name in requireds.keys():
+                    del requireds[pkg.name]
+                del pkgs_map_copy[pkg.name]
+        if without_cyclic_dependencies and len(pkgs_sort) == old_len_sort:
+            break
 
 def generate_target_buildorder(target_path, pkgs_map, fast_build_mode):
     "Generate a build order for building the dependencies of the specified package."
     if target_path.endswith('/'):
         target_path = target_path[:-1]
 
     package_name = os.path.basename(target_path)
-    if "gpkg" in target_path.split("/")[-2].split("-") and "glibc" not in package_name.split("-"):
-        package_name += "-glibc"
+    if "gpkg" in target_path.split("/")[-2].split("-") and not has_prefix_glibc(package_name):
+        package_name = add_prefix_glibc_to_pkgname(package_name)
     package = pkgs_map[package_name]
     # Do not depend on any sub package
     if fast_build_mode:
         package.deps.difference_update([subpkg.name for subpkg in package.subpkgs])
     return package.recursive_dependencies(pkgs_map)
 
+def get_list_cyclic_dependencies(pkgs_map, index=[], ok_pkgs=set(), pkgname=None, build_mode=False):
+    "Find and return circular dependencies for all packages or for one specified package."
+
+    if len(index) == 0:
+        ok_pkgs = {pkg.name for pkg in generate_full_buildorder(pkgs_map, build_mode, True)}
+        range_pkgs = ({pkgname} if pkgname else {pkg for pkg in pkgs_map.keys() if not build_mode or isinstance(pkgs_map[pkg], TermuxPackage)}) - ok_pkgs
+    else:
+        range_pkgs = pkgs_map[index[-1]].deps.copy()
+        if build_mode:
+            for subpkg in pkgs_map[index[-1]].subpkgs:
+                range_pkgs |= subpkg.deps
+            range_pkgs = remove_only_installing_deps(pkgs_map, range_pkgs-{index[-1]}-{subpkg.name for subpkg in pkgs_map[index[-1]].subpkgs})
```

Comment:
Same here, please use spaces around operators so the operands are clearly distinguished.
```suggestion
            range_pkgs = remove_only_installing_deps(pkgs_map, range_pkgs - {index[-1]} - {subpkg.name for subpkg in pkgs_map[index[-1]].subpkgs})
```